docs/blog/2019-01-25-mesh-loadtest.md

```diff
@@ -0,0 +1,82 @@
+# Performance Testing of Service Mesh Receiver
+
+- Auther: Hongtao Gao, Apache SkyWalking & ShardingShpere PMC
+- [GitHub](https://github.com/hanahmily), [Twitter](https://twitter.com/hanahmily), [Linkedin](https://www.linkedin.com/in/gao-hongtao-47b835168/)
+
+Jan. 25th, 2019
+
+Service mesh receiver was first introduced in Apache SkyWalking 6.0.0-beta. It is designed to provide a common entrance for receiving telemetry data from service mesh framework, for instance, Istio, Linkerd, etc. What’s the service mesh? According to Istio’s explain:
+
+The term service mesh is used to describe the network of microservices that make up such applications and the interactions between them.
+
+As a PMC of Apache SkyWalking, I tested trace receiver and well understood the performance of collectors in trace scenario. I also would like to figure out the performance of service mesh receiver.
+
+## Different between trace and service mesh
+
+Following chart presents a typical trace map:
+
+![](/static/blog/2019-01-25-mesh-loadtest/image5.png)
+
+You could find a variety of elements in it just like web service, local method, database, cache, MQ and so on. But service mesh only collect service network telemetry data that contains the entrance and exit data of a service. A smaller quantity of data is sent to the service mesh receiver than the trace.
+
+But using sidecar is a little different.
+
+![](/static/blog/2019-01-25-mesh-loadtest/image1.png)
+
+The client requesting “A” that will send a segment to service mesh receiver from “A”’s sidecar. If “A” depends on “B”,  another segment will be sent from “A”’s sidecar. But for a trace system, only one segment is received by the collector. The sidecar model splits one segment into small segments, that will increase service mesh receiver network overhead.
```

Comment:
If in service mesh mode, we shouldn't call it `segment`, request telemetry data may be better. **Segment** is the segment of a trace.

Reply:
In trace mode, still you will receive segment from both sides. Clien't segment includes the exit span, and Server's segment includes the entry span. Trace scenario always includes more span than mesh. So you need to change this description

Reply:
> In trace mode, still you will receive segment from both sides. Clien't segment includes the exit span, and Server's segment includes the entry span. Trace scenario always includes more span than mesh. So you need to change this description

In this section, I focus on the TCP package. For a single instance, mesh style will send double packages than the trace. More TCP package will affect network traffic(overhead) that means if the amount of 1MB mesh data would be more than 1 MB trace data.